`Plugin-Module` is calculated incorrectly

### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  <!-- Paste your environment details below -->
  ```text
  Paste the output here
  ```

</details>


### What Operating System are you using (both controller, and any agents involved in the problem)?

Linux

### Reproduction steps

* download the latest release of `warnings-ng` plugin and extract its `META-INF\MANIFEST.MF`
* notice that `plugin-Module` is not present



### Expected Results

* `plugin-Module` is set to `plugin`

### Actual Results

the entry is missing

### Anything else?

@uhafner does not [release his plugin](https://github.com/jenkinsci/warnings-ng-plugin/blob/cbfe6631f3f6a84a4b6af794cf9078cc0d5c5fa3/bin/release.sh) in the "normal" way (either `maven-release-plugin` at the root level or [JEP-229](https://github.com/jenkinsci/jep/blob/master/jep/229/README.adoc) ).

this was testing earlier by me when and was working but presumably regressed when the code was [changed to use the maven module path](https://github.com/jenkinsci/maven-hpi-plugin/pull/436/commits/80c9cc33c7dd4e01fb420425dba0af1531548f30).

For the PCT we do not care about where the module is in the reactor - we care about where it is relative to the initial git checkout.

the incorrect location causes failures as the PCT is then expecting `warnings-ng` plugin code to be at the root of the repository (which is not correct).

Causing issues like

```
2023-05-26T16:11:37.061Z] Caused by: java.io.IOException: org.codehaus.plexus.util.xml.pull.XmlPullParserException: Expected root element 'project' but found 'projects' (position: START_TAG seen ...=================================================== -->\n<projects>... @8:11) 
[2023-05-26T16:11:37.061Z] 	at com.cloudbees.hooks.MavenUtils.getModelForPlugin(MavenUtils.java:25)
[2023-05-26T16:11:37.061Z] 	at com.cloudbees.hooks.EnforceMethodFilteringHook.getSurefirePluginVersion(EnforceMethodFilteringHook.java:112)
[2023-05-26T16:11:37.061Z] 	at com.cloudbees.hooks.EnforceMethodFilteringHook.check(EnforceMethodFilteringHook.java:41)
[2023-05-26T16:11:37.061Z] 	... 16 more
[2023-05-26T16:11:37.061Z] Caused by: org.codehaus.plexus.util.xml.pull.XmlPullParserException: Expected root element 'project' but found 'projects' (position: START_TAG seen ...=================================================== -->\n<projects>... @8:11) 
[2023-05-26T16:11:37.061Z] 	at org.apache.maven.model.io.xpp3.MavenXpp3Reader.read(MavenXpp3Reader.java:3929)
```

--- a/src/it/metadata-it/verify.groovy
+++ b/src/it/metadata-it/verify.groovy
@@ -37,7 +37,6 @@ try (JarFile j1 = new JarFile(p1)) {
     assert attributes.getValue('Plugin-ScmConnection').equals('scm:git:https://github.com/jenkinsci/maven-hpi-plugin.git')
     def matcher = attributes.getValue('Plugin-GitHash') =~ pattern
     assert matcher.matches()
-    assert attributes.getValue('Plugin-Module').equals('plugin1')
 }
 
 
@@ -71,8 +70,6 @@ try (JarFile j2 = new JarFile(p2)) {
     assert attributes.getValue('Plugin-ScmConnection').equals('scm:git:https://github.com/jenkinsci/maven-hpi-plugin.git')
     def matcher = attributes.getValue('Plugin-GitHash') =~ pattern
     assert matcher.matches()
-    assert attributes.getValue('Plugin-Module').equals('plugin2')
-
 }
 
 return true;

--- a/src/it/verify-it/verify.groovy
+++ b/src/it/verify-it/verify.groovy
@@ -58,7 +58,6 @@ Files.newInputStream(new File(basedir, 'target/verify-it/META-INF/MANIFEST.MF').
   assert manifest.getMainAttributes().getValue('Plugin-ScmUrl').equals('https://github.com/jenkinsci/verify-it-plugin')
   def matcher = manifest.getMainAttributes().getValue('Plugin-GitHash') =~ pattern
   assert matcher.matches()
-  assert manifest.getMainAttributes().getValue('Plugin-Module') == null
   assert manifest.getMainAttributes().getValue('Plugin-Version').startsWith('1.0-SNAPSHOT')
   assert manifest.getMainAttributes().getValue('Short-Name').equals('verify-it')
   assert manifest.getMainAttributes().getValue('Specification-Title').equals('MyNewPlugin') // was project.description in previous versions, now project.name

--- a/src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractJenkinsManifestMojo.java
+++ b/src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractJenkinsManifestMojo.java
@@ -237,7 +237,6 @@ public abstract class AbstractJenkinsManifestMojo extends AbstractHpiMojo {
         String gitHash = getGitHeadSha1();
         // Deprecated: Use "Implementation-Build" for consistency with core and core components
         addAttributeIfNotNull(mainSection, "Plugin-GitHash", gitHash);
-        addAttributeIfNotNull(mainSection, "Plugin-Module", getModule());
         addAttributeIfNotNull(mainSection, "Implementation-Build", gitHash);
     }
 

